src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -4594,5 +4594,200 @@ class var
             Assert.Equal("var", model.GetSymbolInfo(x3Type).Symbol.ToDisplayString());
             Assert.Null(model.GetAliasInfo(x3Type));
         }
+
+        [Fact]
+        public void SimpleDiscardWithConversion()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        (int _, var x) = (new C(1), 1);
+        (var _, var y) = (new C(2), 2);
+        var (_, z) = (new C(3), 3);
+        System.Console.Write($""Output {x} {y} {z}."");
+    }
+    int _i;
+    public C(int i) { _i = i; }
+    public static implicit operator int(C c) { System.Console.Write($""Converted {c._i}. ""); return 0; }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "Converted 1. Output 1 2 3.");
+        }
+
+        [Fact]
+        public void CannotDeconstructIntoDiscardOfWrongType()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        (int _, string _) = (""hello"", 42);
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics(
+                // (6,30): error CS0029: Cannot implicitly convert type 'string' to 'int'
+                //         (int _, string _) = ("hello", 42);
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""hello""").WithArguments("string", "int").WithLocation(6, 30),
+                // (6,39): error CS0029: Cannot implicitly convert type 'int' to 'string'
+                //         (int _, string _) = ("hello", 42);
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "42").WithArguments("int", "string").WithLocation(6, 39)
+                );
+
+            // TODO REVIEW Should there also be a warning for only using discards (no side-effects possible here)
```

Comment:
From discussion with Vlad and Aleksey, the feeling is that such a warning doesn't add much value to the user. I think that makes sense. Cutting.

---
In reply to: [87700135](https://github.com/dotnet/roslyn/pull/15192#discussion_r87700135) [](ancestors = 87700135)